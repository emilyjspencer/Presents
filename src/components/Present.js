import React, { Component } from 'react';
import { Form, FormGroup, FormControl, FormLabel, Button } from 'react-bootstrap';


class Present extends Component {

    constructor() {
        super();

        this.state = { person: '', present: ''}
    }

    render() {
       return (
         <div>
           <Form>
               <FormGroup>
                   <FormLabel>Name</FormLabel>
                   <FormControl onChange={event => this.setState({ person: event.target.value })} className="name" />
               </FormGroup>
               <FormGroup>
                   <FormLabel>Present</FormLabel>
                   <FormControl onChange={event => this.setState({ present: event.target.value })} className="present" />
               </FormGroup>
           </Form>

         </div>
        )
    }
};

export default Present;