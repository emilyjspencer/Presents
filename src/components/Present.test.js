import React from 'react';
import { shallow } from 'enzyme';
import Present from './Present';


describe('Present', () => {

    let wrapper;

    beforeEach(() => wrapper = shallow(<Present />));

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('initializes a person and present in the component state', () => {
        expect(wrapper.state()).toEqual({ person: '', present: '' });
    });

});

describe('when entering a persons name', () => {

    let wrapper;

    beforeEach(() => wrapper = shallow(<Present />));

    beforeEach(() => {
        wrapper.find('.name').simulate('change', { target: { value: 'Laura' }});
    });
    
    it('updates the person in the component state', () => {
      expect(wrapper.state().person).toEqual('Laura');
    });

});