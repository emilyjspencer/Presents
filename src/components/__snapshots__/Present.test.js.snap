// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Present renders correctly 1`] = `ShallowWrapper {}`;
